sfxclient/sfxclient.go

```diff
@@ -2,226 +2,216 @@ package sfxclient
 
 import (
 	"expvar"
+
+	"github.com/signalfx/golib/datapoint"
+
 	"fmt"
-	"runtime"
 	"sync"
+	"sync/atomic"
 	"time"
 
-	"github.com/Sirupsen/logrus"
-	"github.com/signalfx/golib/datapoint"
-	"github.com/signalfx/golib/logherd"
-
-	// TODO: Figure out a way to not have this dependency
 	"github.com/signalfx/golib/errors"
-	"github.com/signalfx/metricproxy/dp/dpsink"
-	"github.com/signalfx/metricproxy/protocol/signalfx"
+	"github.com/signalfx/golib/timekeeper"
 	"golang.org/x/net/context"
 )
 
-var log *logrus.Logger
+// DefaultReportingDelay is the default interval between new SignalFx schedulers
+const DefaultReportingDelay = time.Second * 20
 
-func init() {
-	log = logherd.New()
+// DefaultErrorHandler is the default way to handle errors by a scheduler.  It simply prints them to stdout
+var DefaultErrorHandler = func(err error) error {
+	fmt.Printf("Unable to handle error: %s", err.Error())
+	return nil
 }
 
-// ClientSink is the expected interface that recieves reported metrics
-type ClientSink interface {
-	dpsink.Sink
-	Endpoint(endpoint string)
-	AuthToken(endpoint string)
+// Sink is anything that can receive points collected by a Scheduler
+type Sink interface {
+	AddDatapoints(ctx context.Context, points []*datapoint.Datapoint) (err error)
```

Comment:
I did not want to create a dependency on the metric proxy code inside Golib

On Mon, Dec 14, 2015, 7:44 PM Matthew Pound <notifications@github.com>
wrote:

> In sfxclient/sfxclient.go
> <https://github.com/signalfx/golib/pull/36#discussion_r47593831>:
>
> >  }
> >
> > -// ClientSink is the expected interface that recieves reported metrics
> > -type ClientSink interface {
> > -	dpsink.Sink
> > -	Endpoint(endpoint string)
> > -	AuthToken(endpoint string)
> > +// Sink is anything that can receive points collected by a Scheduler
> > +type Sink interface {
> > +	AddDatapoints(ctx context.Context, points []*datapoint.Datapoint) (err error)
>
> this is the same as a dpsink.DSink do we need a new interface?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/signalfx/golib/pull/36/files#r47593831>.
>